dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java

```diff
@@ -237,106 +237,122 @@ default List<SourceFile> getAllSourceFiles(long workflowId, String tag, Descript
+
+    /**
+     * Finds the version and entry corresponding to the specified version name and entry ID, adjusting access per the specified logged-in user.
+     * The entry will only contain the version with the specified version name, and will be evicted so no accidental db mods are possible.
+     * Throws a CustomWebApplicationException if the entry or version is not accessible.
+     * @param entryId id of the entry
+     * @param versionName name of the version
+     * @param user the logged-in user
+     * @return the version and parent entry
+     */
+    default EvictedVersionAndEntry<T> findAndCheckVersionByName(long entryId, String versionName, Optional<User> user, VersionDAO versionDAO) {
+        try {
+            String modifiedVersionName = ObjectUtils.firstNonNull(versionName, getDefaultVersionName());
+            versionDAO.enableNameFilter(modifiedVersionName);
+            T entry = findAndCheckEntryById(entryId, user);
+            Version version = entry.getWorkflowVersions().stream().filter(v -> v.getName().equals(modifiedVersionName)).findFirst().orElseThrow(
+                () -> new CustomWebApplicationException("Invalid or missing version " + modifiedVersionName + ".", HttpStatus.SC_BAD_REQUEST));
+            getDAO().evict(entry);
+            return new EvictedVersionAndEntry<>(version, entry);
+        } finally {
+            versionDAO.disableNameFilter();
+        }
+    }
+
+    /**
+     * Finds the version and entry corresponding to the specified version ID and entry ID, adjusting access per the specified logged-in user.
+     * The entry will only contain the version with the specified version ID, and will be evicted so no accidental db mods are possible.
+     * Throws a CustomWebApplicationException if the entry or version is not accessible.
+     * @param entryId id of the entry
+     * @param versionId id of the version
+     * @param user the logged-in user
+     * @return the version and parent entry
+     */
+    default EvictedVersionAndEntry<T> findAndCheckVersionById(long entryId, long versionId, Optional<User> user, VersionDAO versionDAO) {
+        try {
+            versionDAO.enableIdFilter(versionId);
+            T entry = findAndCheckEntryById(entryId, user);
+            Version version = entry.getWorkflowVersions().stream().filter(v -> v.getId() == versionId).findFirst().orElse(null);
+            if (version == null) {
+                throw new CustomWebApplicationException("Version " + versionId + " does not exist for this entry", HttpStatus.SC_BAD_REQUEST);
+            }
+            getDAO().evict(entry);
+            return new EvictedVersionAndEntry<>(version, entry);
+        } finally {
+            versionDAO.disableIdFilter();
+        }
+    }
+
+    /**
+     * Finds the entry corresponding to the specified ID, adjusting access per the specified logged-in user.
+     * The entry will have its versions populated, and will be evicted so no accidental db mods are possible.
```

Comment:
1. This method isn't evicting.
2. Also might want to mention that's it's filtering out hidden versions.

Reply:
Ooops, spurious copy of comment from above, removed.